src/anomalib/models/efficientad/lightning_model.py

```diff
@@ -170,16 +170,43 @@ def map_norm_quantiles(self, dataloader: DataLoader) -> dict[str, Tensor]:
                     map_ae = output["map_ae"]
                     maps_st.append(map_st)
                     maps_ae.append(map_ae)
-        maps_st = torch.cat(maps_st)
-        maps_ae = torch.cat(maps_ae)
-        qa_st = torch.quantile(maps_st, q=0.9).to(self.device)
-        qb_st = torch.quantile(maps_st, q=0.995).to(self.device)
-        qa_ae = torch.quantile(maps_ae, q=0.9).to(self.device)
-        qb_ae = torch.quantile(maps_ae, q=0.995).to(self.device)
+
+        qa_st, qb_st = self.get_quantiles_of_maps(maps_st)
+        qa_ae, qb_ae = self.get_quantiles_of_maps(maps_ae)
         return {"qa_st": qa_st, "qa_ae": qa_ae, "qb_st": qb_st, "qb_ae": qb_ae}
 
+    def _get_quantiles_of_maps(self, maps: list[Tensor]) -> tuple[Tensor, Tensor]:
+        """Calculate 90% and 99.5% quantiles of the given anomaly maps.
+
+        If the total number of elements in the given maps is larger than 16777216
+        the returned quantiles are computed on a random subset of the given
+        elements.
+
+        Args:
+            maps (list[Tensor]): List of anomaly maps.
+
+        Returns:
+            tuple[Tensor, Tensor]: Two scalars - the 90% and the 99.5% quantile.
+        """
+        maps_flat = torch.flatten(torch.cat(maps))
+        # torch.quantile only works with input size up to 16777216 elements
+        # (16777216 is 16 * 1024 * 1024)
+        # if we have more elements we need to decrease the size
+        # we do this by sampling random elements of maps_flat because then
+        # the locations of the quantiles (90% and 99.5%) will still be
+        # valid even though they might not be the exact quantiles.
+        max_input_size = 16777216
```

Comment:
How did you get that number? Is it an arbitrary choice?

Reply:
It is 16 * 1024 * 1024 and is the actual limit of what torch.quantile can handle.

Reply:
Ok, maybe you can use 2**24 here and reference the line in pytorch that causes the limit https://github.com/pytorch/pytorch/blob/b9f81a483a7879cd3709fd26bcec5f1ee33577e6/aten/src/ATen/native/Sorting.cpp#L291

Reply:
Is there a reason you do this check for max input size for quantile and random sample here and not in the forward method of the torch_model.py ?

Reply:
> Ok, maybe you can use 2**24 here and reference the line in pytorch that causes the limit 

Done